packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -427,6 +432,19 @@ class DevFS {
     });
     if (dirtyEntries.isNotEmpty) {
       printTrace('Updating files');
+      if (generator != null) {
+        dirtyEntries.forEach((Uri deviceUri, DevFSContent content) {
+          if (content is DevFSFileContent) {
+            generator.invalidate(content.file.uri);
+          }
+        });
+        final DeltaProgram delta = await generator.computeDelta();
+        final Program program = delta.newProgram;
+        writeProgramToBinary(program, mainPath + ".dill");
```

Comment:
Good question - we can change it to take in a Sink, or make it flexible enough to let you choose.